src/main/java/org/dbos/apiary/xa/BitronixXADBConnection.java

```diff
@@ -0,0 +1,109 @@
+package org.dbos.apiary.xa;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import bitronix.tm.resource.common.TransactionContextHelper;
+import bitronix.tm.resource.jdbc.PoolingDataSource;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import javax.transaction.xa.XAResource;
+
+import java.sql.Connection;
+
+public class BitronixXADBConnection extends BaseXAConnection  {
+    private static final Logger logger = LoggerFactory.getLogger(MySQLXAConnection.class);
```

Comment:
Logger should be `BitronixXADBConnection` here. May have similar issues in other places, please check.